libethereum/BlockChain.cpp

```diff
@@ -1188,38 +1193,55 @@ void BlockChain::garbageCollect(bool _force)
 	m_lastCollection = chrono::system_clock::now();
 
 	Guard l(x_cacheUsage);
-	WriteGuard l1(x_blocks);
-	WriteGuard l2(x_details);
-	WriteGuard l3(x_blockHashes);
-	WriteGuard l4(x_receipts);
-	WriteGuard l5(x_logBlooms);
-	WriteGuard l6(x_transactionAddresses);
-	WriteGuard l7(x_blocksBlooms);
 	for (CacheID const& id: m_cacheUsage.back())
 	{
 		m_inUse.erase(id);
 		// kill i from cache.
 		switch (id.second)
 		{
 		case (unsigned)-1:
+		{
+			WriteGuard l(x_blocks);
 			m_blocks.erase(id.first);
 			break;
+		}
 		case ExtraDetails:
+		{
+			WriteGuard l(x_details);
 			m_details.erase(id.first);
 			break;
+		}
+		case ExtraBlockHash:
+		{
+			WriteGuard l(x_blockHashes);
+			m_blockHashes.erase(static_cast<uint64_t>(FixedHash<32>::Arith(id.first)));
```

Comment:
Sorry, now I see that we actually don't add block hashes into `m_cacheUsage` because of this: https://github.com/ethereum/cpp-ethereum/blob/develop/libethereum/BlockChain.h#L392

So not handling `ExtraBlockHash` was not en error.

Let's remove this code, but leave `case ExtraBlockHash` with a comment that block hashes don't go into `m_cacheUsage`

Reply:
OK.  I'll add a comment.